src/lib/components/Dropdown/DropdownItem.svelte

```diff
@@ -0,0 +1,51 @@
+<script lang="ts">
+	import {
+		dropdownItemVariants,
+		type DropdownItemProps
+	} from '../../../styles/tailwind/dropdownItem';
+
+	export let text: string;
+	export let disabled: boolean = false;
+	export let currentSelectedChoice: string | null = null;
+
+	let currentVariant: DropdownItemProps['variant'] = 'default';
+	let isHover: boolean = false;
+	let isSelected = false;
+
+	$: isSelected = currentSelectedChoice === text;
+
+	$: {
+		if (disabled) {
+			currentVariant = 'disabled';
+		} else if (isHover) {
+			currentVariant = 'hover';
+		} else if (isSelected) {
+			currentVariant = 'pressed';
+		} else {
+			currentVariant = 'default';
+		}
+	}
+
+	function handleClick() {
+		isHover = false;
+		currentSelectedChoice = text;
+	}
```

Comment:
we can just emit event instead of relying on two way binding
```svelte
function handleClick() {
  if (!disabled) {
    dispatch('select', text);
  }
}
```


dont forget to import this on top

```typescript
import { createEventDispatcher } from 'svelte';
const dispatch = createEventDispatcher();
```